packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -742,12 +743,12 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         super.dispose();
     }
 
-    protected resizeTerminal(): void {
+    protected resizeTerminal = debounce((): void => {
```

Comment:
Suggestion: To facilitate simple overrides of this method, we usually follow a `<method> = debounce(do<method>)` pattern. See [here](https://github.com/eclipse-theia/theia/blob/de84d3aec991f65338dbdb1397d3c9948a684dde/packages/core/src/browser/theming.ts#L88-L98) for an example. Please extract this into a `doResizeTerminal` method.